### ISSUE TITLE
`clippy::transmute_float_to_int` in const context should not trigger when rust-version is less than 1.83 in Cargo.toml 

### ISSUE THREAD

**Issue by tspiteri** [rust-lang/rust-clippy#13383]
### Summary

When rust-version is less than 1.83, the suggested code does not compile. 

### Lint Name

clippy::transmute_float_to_int

### Reproducer

Cargo.toml contains:
```toml
rust-version = "1.82"
```

I tried this code:

```rust
pub const fn f32_to_u32(f: f32) -> u32 {
    unsafe { core::mem::transmute(f) }
}
```

I saw this happen:

```
warning: transmute from a `f32` to a `u32`
 --> src/lib.rs:2:14
  |
2 |     unsafe { core::mem::transmute(f) }
  |              ^^^^^^^^^^^^^^^^^^^^^^^ help: consider using: `f.to_bits()`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#transmute_float_to_int
  = note: `#[warn(clippy::transmute_float_to_int)]` on by default
```

I expected to see this happen:
No warning as rust 1.82.0 does not support `f32::to_bits` in const context.

### Version

```text
rustc 1.83.0-nightly (0ee7cb5e3 2024-09-10)
binary: rustc
commit-hash: 0ee7cb5e3633502d9a90a85c3c367eccd59a0aba
commit-date: 2024-09-10
host: x86_64-unknown-linux-gnu
release: 1.83.0-nightly
LLVM version: 19.1.0
```

**Comment by y21** [comment]
`transmute_float_to_int` already checks if it's in a const context, so it shouldn't fire here regardless of the version:
https://github.com/rust-lang/rust-clippy/blob/5ead90f13ae3e03f0c346aea3198f18298960d83/clippy_lints/src/transmute/transmute_float_to_int.rs#L21

The reproducer also doesn't have any warnings: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=87b21a87034fd451c9278971fe6971bb

**Comment by tspiteri** [comment]
In rust 1.83, `f32::to_bits` will be usable on const context.

**Comment by tspiteri** [comment]
It seems that in the rust-clippy repo it is conditioned on `const_context` as pointed out above,but in the rust repo it is not:

https://github.com/rust-lang/rust/blob/1f51450c6826cc4c3d1089cd3f57089a14f49190/src/tools/clippy/clippy_lints/src/transmute/transmute_float_to_int.rs#L20

**Comment by y21** [comment]
I see. Yeah, looks like those checks were removed in rust-lang/rust#129555 which was merged after the last rust->clippy sync, so that makes sense for why the bug isn't present in the clippy repo yet

**Comment by tspiteri** [comment]
Since this is in the rust repo, I created https://github.com/rust-lang/rust/pull/130305 in the rust repo for it.
